newscript.go

```diff
@@ -134,141 +162,83 @@ func goCompatibility() error {
 
 func goGet() error {
 	fmt.Printf("--------Getting u-root \n")
-	cmd1 := exec.Command("go", "get", "github.com/u-root/u-root")
-	err := cmd1.Run()
-	//TODO: Figure why this ^ isn't working
+	cmd := exec.Command("sh", "-c", "export", "GOPATH=\"$HOME/go\"")
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("exported \n")
+	cmd = exec.Command("go", "get", "github.com/u-root/u-root/")
+	err = cmd.Run()
 	/*if err != nil {
 		return err
 	}*/
-	fmt.Printf("--------Gotu-root \n")
+	fmt.Printf("--------Got u-root \n")
 	gopath := fmt.Sprintf("GOPATH=%s/go", homeDir)
-	bbpath := fmt.Sprintf("GOPATH=%s/go/src/github.com/u-root/u-root/bb/bb", homeDir)
-	cmd3 := exec.Command(gopath, bbpath)
-	err = cmd3.Run()
+	bbpath := fmt.Sprintf("%s/go/src/github.com/u-root/u-root/bb/bb", homeDir)
+	cmd = exec.Command("go", "build", "bbpath")
+	err = cmd.Run()
 	if err != nil {
 		return err
 	}
-	fmt.Printf("--------Getting bb \n")
-/* Removing in favor of bb
-	cmd4 := exec.Command("go", "run", "scripts/ramfs.go")
-	err = cmd4.Run()
+	cmd = exec.Command(gopath, bbpath)
+	err = cmd.Run()
 	if err != nil {
 		return err
 	}
-*/
+	fmt.Printf("--------Getting bb \n")
 	if _, err := os.Stat("/tmp/initramgs.linux_amd64.cpio"); err != nil {
 		return err
 	}
 	fmt.Printf("Created the initramfs in /tmp/")
 	return nil
 }
 
-// Get the right kernel for the current config from github (needs work)
-// https://github.com/LaconicPneumonic/linux-4.12.7.git
-func kernelGet() error {
-	fmt.Printf("-------- Getting the kernel \n")
-	cmd8 := exec.Command("git", "clone", "https://github.com/LaconicPneumonic/linux-4.12.7.git")
-	err := cmd8.Run()
-	if err != nil {
-		fmt.Printf("didn't clone kernel from Anthony's repo")
-		return err
-	}
-	return nil
-}
 
-
-
-// Get the right kernel for the current config in NiChrome (Needs work)
-func linuxKernelGet() error {
+func kernelGet() error {
 	fmt.Printf("-------- Getting the kernel \n")
-	fmt.Printf("Current version of Kernel is %v\n", kernelVersion)
-	name := fmt.Sprintf("%s.tar", linuxVersion)
-	out, err := os.Create(name)
-	fmt.Printf("Kernel saved in %s\n", name)
-	if err != nil {
-		return err
-	}
-	defer out.Close()
-	linkName := fmt.Sprintf("%s%s%s", "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-", kernelVersion, ".tar.xz")
-	fmt.Printf("Pulling from link %s\n", linkName)
-	resp, err := http.Get(linkName)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-	_, err = io.Copy(out, resp.Body)
-	if err != nil {
-		return err
-	}
-	cmd5 := exec.Command("tar", "xf", fmt.Sprintf("%s.tar",linuxVersion))
-	err = cmd5.Run()
+	cmd := exec.Command("git", "clone","-b", "v4.12.7", "git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git")
```

Comment:
did the depth 1 fail here? Or just on the other place? If it failed in one place, it doesn't make sense to remove it everywhere, right? I'll test this.